src/sdk/Weapons/Projectile.js

```diff
@@ -3,35 +3,44 @@ import chebyshev from "chebyshev";
 import { Weapon } from "./Weapon";
 
 export class Projectile {
+
   /*
     This should take the player and mob object, and do chebyshev on the size of them
   */
-  constructor(damage, from, to, attackStyle, forceSWOnly) {
+  constructor(damage, from, to, attackStyle, forceSWOnly = false, image = null) {
     this.damage = Math.floor(damage);
     if (this.damage > to.currentStats.hitpoint){
       this.damage = to.currentStats.hitpoint;
     }
     this.fromLocation = from.location;
     this.toLocation = to.location;
+    // for visual purposes only
+    this.fromSize = from.size;
```

Comment:
Might as well just attach the entire `from` and `to` to the projectile rather than copying over individual properties

Reply:
It's an interesting problem: most projectiles actually seek towards a target, so `to` make sense for that. There are some projectiles that just go towards an area instead of a unit though, like webs, xarpus splat, vorkath fireballs. Should they both be represented by this class?

Reply:
My first thought is a subclass or something called "AreaProjectile" that can take a region+point combo as a "to"... But I haven't thought that through thoroughly

Reply:
I will consider this more though

Reply:
So after some thought, I agree that projectiles should belong to the region and not the mobs or player. 

I think the projectile class should have a super class, and then a new area projectile created that subclasses that new super class. Add it to the region and let the region handle the logic of transferring splats to the correct player or mob